server.py

```diff
@@ -58,14 +59,19 @@ def get_response_stream():
         default="musabgultekin/functionary-7b-v1",
         help="Model name",
     )
+    parser.add_argument(
+        "--device",
+        type=str,
+        default="auto",
+        help="choose which device to host the model: cpu, cuda, cuda:xxx, or auto",
+    )
     parser.add_argument("--load_in_8bit", type=bool, default=False)
     args = parser.parse_args()
-
     model = LlamaForCausalLM.from_pretrained(
         args.model,
         low_cpu_mem_usage=True,
-        device_map="auto",
-        torch_dtype=torch.float16,
+        device_map=args.device,
+        torch_dtype=torch.bfloat16,
```

Comment:
Was there a reason for this dtype change? Cause blfoat16 was problematic on some cases. and float16 was working much better on inference